src/sql/src/plan/statement/ddl.rs

```diff
@@ -2992,27 +3001,129 @@ pub fn plan_create_connector(
         if_not_exists,
     } = stmt;
     let connector_inner = match connector {
-        CreateConnector::Kafka {
-            broker,
-            with_options,
-        } => {
-            let mut with_options = normalize::options(&with_options);
-            ConnectorInner::Kafka {
-                broker: broker.parse()?,
-                config_options: kafka_util::extract_config(&mut with_options)?,
-            }
-        }
+        CreateConnector::Kafka { broker, security } => ConnectorInner::Kafka {
+            broker: broker.parse()?,
+            // TODO resolve secrets from names to values
+            security: match security {
+                mz_sql_parser::ast::KafkaSecurityOptions::SSL {
+                    key,
+                    certificate,
+                    passphrase,
+                    authority,
+                } => {
+                    let key_id = if let Some(key) = key.as_ref() {
```

Comment:
My dev machine is down atm, but this looks like it wants to use the more idiomatic `map` operation here, i.e.

```rust
let key_id = key.map(|key| {
  let key_id = name_to_id(scx, key)?;
  depends_on.push(key_id);
  key_id
}
```

Reply:
Using map the thing I cannot figure out a reasonable approach for is handling the failure case of `name_to_id` since the `?` does not work in the closure because the closure has to return an `Option`. This is what led me to use the if/let pattern originally because it made cleanly propagating that error convenient.

Reply:
Here's a better version:

```diff
diff --git a/src/sql/src/plan/statement/ddl.rs b/src/sql/src/plan/statement/ddl.rs
index 53d9d434b..22fc94a0a 100644
--- a/src/sql/src/plan/statement/ddl.rs
+++ b/src/sql/src/plan/statement/ddl.rs
@@ -3011,39 +3011,22 @@ pub fn plan_create_connector(
                     passphrase,
                     authority,
                 } => {
-                    let key_id = if let Some(key) = key.as_ref() {
-                        let key_id = name_to_id(scx, key)?;
-                        depends_on.push(key_id);
-                        Some(key_id)
-                    } else {
-                        None
-                    };
-                    let certificate_id = if let Some(certificate) = certificate.as_ref() {
-                        let certificate_id = name_to_id(scx, certificate)?;
-                        depends_on.push(certificate_id);
-                        Some(certificate_id)
-                    } else {
-                        None
-                    };
-                    let passphrase_id = if let Some(passphrase) = passphrase.as_ref() {
-                        let passphrase_id = name_to_id(scx, passphrase)?;
-                        depends_on.push(passphrase_id);
-                        Some(passphrase_id)
-                    } else {
-                        None
-                    };
-                    let authority_id = if let Some(authority) = authority.as_ref() {
-                        let authority_id = name_to_id(scx, authority)?;
-                        depends_on.push(authority_id);
-                        Some(authority_id)
-                    } else {
-                        None
-                    };
+                    let mut f =
+                        |v: Option<UnresolvedObjectName>| -> Result<Option<String>, anyhow::Error> {
+                            v.map_or_else(
+                                || Ok(None),
+                                |i| {
+                                    let id = name_to_id(scx, &i)?;
+                                    depends_on.push(id);
+                                    Ok(Some(id.to_string()))
+                                },
+                            )
+                        };
                     KafkaSecurityOptions::SSL {
-                        key: key_id.map_or_else(|| None, |i| Some(i.to_string())),
-                        certificate: certificate_id.map_or_else(|| None, |i| Some(i.to_string())),
-                        passphrase: passphrase_id.map_or_else(|| None, |i| Some(i.to_string())),
-                        authority: authority_id.map_or_else(|| None, |i| Some(i.to_string())),
+                        key: f(key)?,
+                        certificate: f(certificate)?,
+                        passphrase: f(passphrase)?,
+                        authority: f(authority)?,
                     }
                 }
```

Reply:
Rather than using map on the options I opted to just write a small function for this since that makes the error semantics nice and clear while still eliminating all of the duplication and boilerplate.